agent/ecscni/netconfig_linux.go

```diff
@@ -169,3 +171,46 @@ func NewAppMeshConfig(appMesh *appmesh.AppMesh, cfg *Config) (string, *libcni.Ne
 
 	return defaultAppMeshIfName, networkConfig, nil
 }
+
+// NewServiceConnectNetworkConfig creates a new ServiceConnect CNI network configuration
+func NewServiceConnectNetworkConfig(
+	scConfig *serviceconnect.Config,
+	enableIPv4, enableIPv6 bool,
+	cfg *Config) (string, *libcni.NetworkConfig, error) {
+	var ingressConfig []IngressConfigJSONEntry
+	for _, ic := range scConfig.IngressConfig {
+		newEntry := IngressConfigJSONEntry{
+			ListenerPort: ic.ListenerPort,
+		}
+		if ic.InterceptPort != nil {
+			newEntry.InterceptPort = *ic.InterceptPort
+		}
+		ingressConfig = append(ingressConfig, newEntry)
+	}
+
+	var egressConfig *EgressConfigJSON
+	if scConfig.EgressConfig != nil {
+		egressConfig = &EgressConfigJSON{
+			ListenerPort: scConfig.EgressConfig.ListenerPort,
+			VIP: VIPConfigJSON{
+				IPv4CIDR: scConfig.EgressConfig.VIP.IPV4CIDR,
+				IPv6CIDR: scConfig.EgressConfig.VIP.IPV6CIDR,
+			},
+		}
+	}
+
+	scNetworkConfig := ServiceConnectConfig{
+		Name:          ECSServiceConnectPluginName,
+		Type:          ECSServiceConnectPluginName,
+		IngressConfig: ingressConfig,
+		EgressConfig:  egressConfig,
+		EnableIPv4:    enableIPv4,
+		EnableIPv6:    enableIPv6,
+	}
+	networkConfig, err := newNetworkConfig(scNetworkConfig, ECSServiceConnectPluginName, cfg.MinSupportedCNIVersion)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "NewServiceConnectNetworkConfig: construct the service connect network configuration failed")
```

Comment:
Borrow wisdom from Angel ([source](https://github.com/aws/amazon-ecs-agent/pull/3208#discussion_r870645566))
>nit: the preferred way to wrap errors is now fmt.Errorf("message: %w", err)
github.com/pkg/errors is a bit of a legacy package now

Reply:
Sure will address in next rev